dascore/utils/remote_io.py

```
@@ -135,12 +135,24 @@ def _download_remote_file(path, local_path: Path):
-            resource.open("rb", **open_kwargs) as remote_fi,
-            tmp_path.open("wb") as local_fi,
-        ):
-            while chunk := remote_fi.read(get_config().remote_download_block_size):
-                local_fi.write(chunk)
+        if protocol in _HTTP_PROTOCOLS:
+            # Use a direct blocking HTTP download here rather than
+            # ``resource.open(...)``. The fallback path can be entered while an
+            # active fsspec HTTP read is already in progress, and re-entering
+            # that stack from inside the fallback can deadlock.
+            headers = dict(getattr(resource, "storage_options", {}) or {})
+            request = Request(str(resource), headers=headers)
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)